sdks/go/pkg/beam/runners/universal/universal.go

```diff
@@ -82,6 +82,54 @@ func Execute(ctx context.Context, p *beam.Pipeline) error {
 		return errors.WithContextf(err, "generating model pipeline")
 	}
 
+	// Adding Expanded transforms to their counterparts in the Pipeline
+	for id, external := range p.ExpandedTransforms {
+		pipeline.Requirements = append(pipeline.Requirements, external.Requirements...)
+
+		// Correct update of transform corresponding to the ExpandedTransform
+		// TODO(pskevin): Figure if there is a better way of supporting multiple outputs
+		transform := pipeline.Components.Transforms[id]
+		existingInput := ""
+		newInput := ""
+		for _, v := range transform.Outputs {
+			existingInput = v
+		}
+		for _, v := range external.ExpandedTransform.Outputs {
+			newInput = v
+		}
+
+		for _, t := range pipeline.Components.Transforms {
+			for idx, i := range t.Inputs {
+				if i == existingInput {
+					t.Inputs[idx] = newInput
+				}
+			}
+		}
+
+		// Adding components of the Expanded Transforms to the current Pipeline
+		for k, v := range external.Components.Transforms {
```

Comment:
You should be able to clean this up with pipelinex.Update

https://github.com/apache/beam/blob/master/sdks/go/pkg/beam/core/runtime/pipelinex/replace.go#L32

Note that you'll need to delete the "go" entry from the environments first to do that safely for the loopback mode fix.

Reply:
That's an awesome helper. Thanks for pointing it out.